File: problems/237/Lc237.c
/**
 * Definition for singly-linked list.
 */
struct ListNode {
  int val;
  struct ListNode* next;
};

void deleteNode(struct ListNode* node) {
  // 1.
  // node->val = node->next->val;
  // node->next = node->next->next;

  // 2.
  struct ListNode* next_node = node->next;
  node->val = next_node->val;
  node->next = next_node->next;
}

int main() {
  return 0;
}

File: problems/237/Lc237.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    def deleteNode(self, node):
        # 1.
        # prev = None
        # while node.next:
        #     node.val = node.next.val
        #     prev = node
        #     node = node.next
        # prev.next = None

        # 2.
        # node.val = node.next.val
        # node.next = node.next.next

        # 3.
        # node.val, node.next = node.next.val, node.next.next

        # 4.
        # nextNode = node.next
        # node.val = nextNode.val
        # node.next = nextNode.next

        # 5.
        nextNode = node.next
        node.val, node.next = nextNode.val, nextNode.next
